src/core/monitor/gnss_synchro_monitor.cc

```diff
@@ -55,11 +55,12 @@ gnss_synchro_monitor::gnss_synchro_monitor(int n_channels,
 
 void gnss_synchro_monitor::forecast(int noutput_items __attribute__((unused)), gr_vector_int& ninput_items_required)
 {
-    for (int32_t channel_index = 0; channel_index < d_nchannels; channel_index++)
+    for (int32_t channel_index = 0; channel_index < d_nchannels - 1; channel_index++)
```

Comment:
a comment above the block would be helpful; your change is not intuitive and a subsequent maintainer might not understand it.

I see the issue; by returning 0, it can cause this to spin when no samples are available (although the scheduler should still only be calling every 250ms). On the other hand, this says that a sample **must** be available in input channel `N-1` ... is that guaranteed? all other inputs would be starved until this was satisfied, if I understand the forecast method properly.

Reply:
I've taken the logic from observables block, that does not hog the CPU when there are no synchros to process:
https://github.com/gnss-sdr/gnss-sdr/blob/next/src/algorithms/observables/gnuradio_blocks/hybrid_observables_gs.cc#L493-L501

Reply:
I guess it's a pattern. Yeah, add the comments and it is fine.